### PR TITLE
Add PhoneNumber type

### PR DESCRIPTION
**Pull request by garry-jeromson**
## Change Summary

Adds a PhoneNumber type, validated using the phonenumbers (https://pypi.org/project/phonenumbers/) library.

## Checklist

* [x] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Humm I'm not sure about this.

Please create an issue to discuss this and get feedback from others before we do more work on this PR.

**Comment by PrettyWood** [comment]
@garry-jeromson @samuelcolvin The related issue has quite a lot of success. I reckon we should keep working on this then

**Comment by garry-jeromson** [comment]
Yeah, I'm game. The only outstanding question for me is the generally strategy/approach when it comes to using third-party libraries for data validation; what criteria need to be fulfilled, what should be avoided, etc.

File: changes/1550-garry-jeromson.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add a PhoneNumber type, validated using the phonenumbers (https://pypi.org/project/phonenumbers/) library.
```
Comment (samuelcolvin) [inline changes/1550-garry-jeromson.md:1]: markdown please.

File: docs/usage/types.md [hunk 2]
```diff
@@ -689,6 +692,14 @@ The actual validation verifies the card number is:
 * [luhn](https://en.wikipedia.org/wiki/Luhn_algorithm) valid
 * the correct length based on the BIN, if Amex, Mastercard or Visa, and between
   12 and 19 digits for all other brands
+  
+### Phone Numbers
+Requires [phonenumbers](https://pypi.org/project/phonenumbers/) to be installed.
+
+The `PhoneNumber` type validates phone numbers in [E.164 format](https://en.wikipedia.org/wiki/E.164); the output is a 
+simple string.
+
```
Comment (samuelcolvin) [inline docs/usage/types.md:701]: this needs an example.

File: pydantic/types.py [hunk 3]
```diff
@@ -783,3 +787,42 @@ def to(self, unit: str) -> float:
...
+
+    @classmethod
+    def validate(cls, value: Union[str]) -> str:
+        if phonenumbers is None:
+            import_phonenumbers()
+
+        phone_number = value.strip()
+
+        try:
+            phonenumbers.parse(phone_number)
```
Comment (samuelcolvin) [inline pydantic/types.py:825]: See my comment, https://github.com/samuelcolvin/pydantic/issues/1551#issuecomment-687820464. We need to decide about how this should work before continuing.

**Review by samuelcolvin — COMMENTED** [review]
Good start but there's more to do, I'll review properly once we've decide how to proceed and the PR is more complete.

**Comment by samuelcolvin** [comment]
closing this until we have some input from others on how phone numbers should work.

Happy to re-open once we agree or consider a new PR.

**Pull request closed without merging**